schlib/schlib.py

```diff
@@ -363,25 +364,26 @@ def libToDcmFilename(self,filename):
         return os.path.join(dir_path, filename[0] + '.dcm')
 
     def __parse(self):
-        f = open(self.filename, 'r')
+        f = open(self.filename, 'rU')
```

Comment:
Python 2 is not dead, it's (unfortunately) the default Python on macOS :) Granted, I agree that it is oooold... and this rule won't run unless you change the platform check to "darwin".

Are we only guaranteeing these scripts will work on Python 3, then?

Reply:
python 2 IS dead. No matter what some people say it is no longer developed. And only supported till the end of 2019 see: https://pythonclock.org/ *

In my opinion we therefore have till end of 2019 to transfer all our scripts to python 3. (Luckily right now all our scripts seem to run reliably in python 3 so i see no problem with that.)

Every operating system i know of at least offers python 3 as an alternative. So there really is no point in investing any time in keeping our scripts valid for version 2 at this point in time. (Especially if python 3 offers nicer alternatives to a problem.)

So my suggestion would be to mark the files as python 3 by adding `#!/usr/bin/env python3` as the first line and drop support for version 2 as it is simply holding us back.

---
*) _Most linux distros are expected to switch the default python environment sooner. This is because they can not rely on a dead thing in long term support releases. (They already know that there will not be any security updates for python 2 during the life time of their operating system.)_

_Part of kicad 5.1 development was focused on getting kicad compatible with python 3. So 5.1 will be published with python 3 at least in some linux distros. (It is a compiler option now.) I hope kicad does not ship with pyhton 2 as soon as it is not supported any more._

Reply:
> No matter what some people say it is no longer developed.

I didn't say that either, only that macOS comes with it out of the box.